### PR TITLE
Initial Implementation

### LINKED ISSUE #1: Fluent API for ProcessBuilder
We need fluent APIs for:

- all settable properties of `ProcessStartInfo`
- There should be an option to register an `ILogger` from `Microsoft.Extensions.Logging.Abstractions` which will automatically log standard and error output
- There should also be an option to pass a delegate that will be called when standard and/or error output are streamed (see `OutputDataReceived` and `ErrorDataReceived` event)

Additionally to CreateProcess, there should also be a `RunProcessAsync` method that immediately executes the process, taking into account all the settings that were passed to the `ProcessBuilder`.

### PR DESCRIPTION
**Pull request by feO2x**
- introduced `ProcessBuilder` class that allows to fluently configure a `ProcessStartInfo` as well as logging and valid exit codes
- the `FluentProcess` class is essentially a wrapper around a `Process` instance that handles logging and exit code verification

### CODE REVIEW

**Comment by feO2x** [comment]
@andreas-koenig I've also added the ability to add custom handlers for `Process.OutputDataReceived` and `Process.ErrorDataReceived` as well as automatic exit code logging.

**Comment by feO2x** [comment]
@andreas-koenig The readme is now ready, too: https://github.com/Synnotech-AG/Synnotech.FluentProcesses/tree/issue-1

File: Code/Synnotech.FluentProcesses/Extensions.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            StandardOutputEncoding = processStartInfo.StandardOutputEncoding,
+            UseShellExecute = processStartInfo.UseShellExecute,
+            ErrorDialogParentHandle = processStartInfo.ErrorDialogParentHandle,
+        };
+
+#if NET6_0
+        if (OperatingSystem.IsWindows())
+            CopyWindowsSpecificProperties(processStartInfo, clone);
+#else
+        CopyWindowsSpecificProperties(processStartInfo, clone);
```
Comment (andreas-koenig) [inline Code/Synnotech.FluentProcesses/Extensions.cs]: If the target is .NET Standard 2.0, couldn't we run on Linux?

Comment (feO2x) [reply Code/Synnotech.FluentProcesses/Extensions.cs]: Only if used with Mono. However, there is no .NET Standard API to check which platform you are running on, as far as I remember. But we could make this configurable.

Comment (feO2x) [reply Code/Synnotech.FluentProcesses/Extensions.cs]: Hold on - I just found this:
https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.runtimeinformation.osdescription?view=net-6.0#system-runtime-interopservices-runtimeinformation-osdescription

**Pull request merged**